core/src/main/java/org/elasticsearch/indices/IndicesWarmer.java

```diff
@@ -100,27 +94,11 @@ private void warmInternal(final WarmerContext context, boolean topReader) {
         }
         indexShard.warmerService().onPreWarm();
         long time = System.nanoTime();
-        final List<TerminationHandle> terminationHandles = Lists.newArrayList();
-        // get a handle on pending tasks
         for (final Listener listener : listeners) {
             if (topReader) {
-                terminationHandles.add(listener.warmTopReader(indexShard, indexMetaData, context, threadPool));
+                listener.warmTopReader(indexShard, indexMetaData, context);
             } else {
-                terminationHandles.add(listener.warmNewReaders(indexShard, indexMetaData, context, threadPool));
-            }
-        }
-        // wait for termination
-        for (TerminationHandle terminationHandle : terminationHandles) {
-            try {
-                terminationHandle.awaitTermination();
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                if (topReader) {
-                    logger.warn("top warming has been interrupted", e);
-                } else {
-                    logger.warn("warming has been interrupted", e);
-                }
-                break;
+                listener.warmNewReaders(indexShard, indexMetaData, context);
```

Comment:
Right, and current changes will hopefully make warming faster, like doc-values by default (ES 2.0) or disk-based norms (Lucene 5.3).